### PR TITLE
G2P-194 - Create Terminology documentation

### PR DESCRIPTION
**Pull request by olaaustine**

### CODE REVIEW

File: src/views/TerminologyView.vue [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+          <br />
+          <section id="variant-consequences">
+            <h4>Variant Consequences</h4>
+            <div class="pt-3">
+              <table class="table">
+                <thead>
+                  <tr>
+                    <th>Primary Consequence</th>
+                    <th>Variant Consequence</th>
+                    <th>Description in SO</th>
```
Comment (dglemos) [inline src/views/TerminologyView.vue:302]: `Description in Sequence Ontology` is clear

File: src/views/TerminologyView.vue [hunk 2]
```diff
@@ -0,0 +1,348 @@
...
+                      v-for="(term, termIndex) in VariantDescriptionData[
+                        consequences
+                      ]"
+                      :key="term.accession"
+                    >
+                      <td
+                        v-if="termIndex === 0"
+                        :rowspan="VariantDescriptionData[consequences].length"
+                      >
+                        <strong>{{ consequences }}</strong>
```
Comment (dglemos) [inline src/views/TerminologyView.vue:324]: Only the headers should use bold.

File: src/views/TerminologyView.vue [hunk 3]
```diff
@@ -0,0 +1,348 @@
...
+            </div>
+          </section>
+          <br />
+          <section id="molecular-mechanism">
+            <h4>Molecular Mechanism</h4>
+            <div class="pt-3">
+              <table class="table">
+                <thead>
+                  <tr>
+                    <th>Consequence</th>
```
Comment (dglemos) [inline src/views/TerminologyView.vue:236]: Consequence is not the best term, maybe `mechanism`?

File: src/views/TerminologyView.vue [hunk 4]
```diff
@@ -0,0 +1,348 @@
...
+            </div>
+          </section>
+          <br />
+          <section id="mechanism-synopsis">
+            <h4>Molecular Mechanism Synopsis</h4>
+            <div class="pt-3">
+              <table class="table">
+                <thead>
+                  <tr>
+                    <th>Consequence</th>
```
Comment (dglemos) [inline src/views/TerminologyView.vue:268]: The same here. Maybe `mechanism synopsis`?

File: src/views/TerminologyView.vue [hunk 5]
```diff
@@ -0,0 +1,348 @@
...
+                  </tr>
+                </thead>
+                <tbody>
+                  <template
+                    v-for="con_category in terminologyDescriptionData.confidence_category"
+                  >
+                    <tr
+                      v-for="(description, category) in con_category"
+                      :key="category"
+                    >
```
[the remark this answers is not in the document]
Comment (seeta-ramaraju) [reply src/views/TerminologyView.vue:151]: For the terminologyDescriptionData, each specific data type (for eg. terminologyDescriptionData.confidence_category) is a single level array so a single v-for is sufficient. Please make these changes for all types of data in terminologyDescriptionData)

File: src/views/TerminologyView.vue [hunk 6]
```diff
@@ -0,0 +1,348 @@
...
+                  </tr>
+                </thead>
+                <tbody>
+                  <template
+                    v-for="consequences in MolecularDescriptionData.mechanism"
+                  >
+                    <tr
+                      v-for="(description, consequence) in consequences"
+                      :key="consequence"
+                    >
```
Comment (seeta-ramaraju) [inline src/views/TerminologyView.vue:247]:
```suggestion
                    <tr
                      v-for="(description, consequence) in MolecularDescriptionData.mechanism"
                      :key="consequence"
                    >
```

File: src/views/TerminologyView.vue [hunk 7]
```diff
@@ -0,0 +1,348 @@
...
+                  </tr>
+                </thead>
+                <tbody>
+                  <template
+                    v-for="consequences in MolecularDescriptionData.mechanism_synopsis"
+                  >
+                    <tr
+                      v-for="(description, consequence) in consequences"
+                      :key="consequence"
+                    >
```
Comment (seeta-ramaraju) [inline src/views/TerminologyView.vue:279]:
```suggestion
                    <tr
                      v-for="(description, consequence) in MolecularDescriptionData.mechanism_synopsis"
                      :key="consequence"
                    >
```

File: src/views/TerminologyView.vue [hunk 8]
```diff
@@ -0,0 +1,348 @@
...
+                        {{ consequences }}
+                      </td>
+                      <td>{{ term.term }}</td>
+                      <td>
+                        <a
+                          :href="
+                            'http://www.sequenceontology.org/browser/current_release/term/' +
+                            term.accession
+                          "
+                          target="_blank"
```
Comment (seeta-ramaraju) [inline src/views/TerminologyView.vue:333]:
```suggestion
                          :href="`http://www.sequenceontology.org/browser/current_release/term/${term.accession}`"
                          style="text-decoration: none"
                          target="_blank"
```

File: src/views/TerminologyView.vue [hunk 9]
```diff
@@ -0,0 +1,344 @@
...
+            >
+          </li>
+          <li class="nav-item">
+            <a class="nav-link text-dark" href="#variant-consequences"
+              >Variant Consequences</a
+            >
+          </li>
+        </ul>
+      </div>
+    </div>
```
Comment (seeta-ramaraju) [inline src/views/TerminologyView.vue:117]: We can fix this navigation issue in a separate PR. For now, please remove this block of code.

Comment (olaaustine) [reply src/views/TerminologyView.vue]: Done

File: src/views/TerminologyView.vue [hunk 10]
```diff
@@ -0,0 +1,302 @@
...
+          this.isDataLoading = false;
+          this.errorMsg = error.message;
+          console.log(error);
+        });
+    },
+  },
+};
+</script>
+<template>
+  <div class="container-fluid d-flex">
```
Comment (seeta-ramaraju) [inline src/views/TerminologyView.vue:80]: This outer div is not required since the div in line 81 takes care of it

**Review by seeta-ramaraju — APPROVED** [review]
LGTM

**Review by dglemos — APPROVED** [review]

**Pull request merged**